src/histolab/tiler.py

```diff
@@ -100,19 +122,62 @@ def locate_tiles(
         img.putalpha(alpha)
         draw = PIL.ImageDraw.Draw(img)
 
-        tiles = (
-            self._tiles_generator(slide, extraction_mask)[0]
-            if isinstance(self, ScoreTiler)
-            else self._tiles_generator(slide, extraction_mask)
-        )
+        if tiles is None:
+            tiles = (
+                self._tiles_generator(slide, extraction_mask)[0]
+                if isinstance(self, ScoreTiler)
+                else self._tiles_generator(slide, extraction_mask)
+            )
         tiles_coords = (tile[1] for tile in tiles)
-        for coords in tiles_coords:
+
+        for coords, one_outline in self._tile_coords_and_outline_generator(
+            tiles_coords, outline
+        ):
             rescaled = scale_coordinates(coords, slide.dimensions, img.size)
-            draw.rectangle(tuple(rescaled), outline=outline)
+            draw.rectangle(tuple(rescaled), outline=one_outline, width=linewidth)
         return img
 
     # ------- implementation helpers -------
 
+    @staticmethod
+    def _tile_coords_and_outline_generator(
```

Comment:
FYI: We decided that all methods and properties (public or private) should be A-Z ordered so please keep in mind that all new method should be placed in the right position :D

Reply:
@ernestoarbitrio OK, addressed.